### PR TITLE
Data Exports refactoring

### PR DESCRIPTION
**Pull request by viceromag**
This rearranges user-related fields in DataExport and removes scheduling endpoint.

PR for adding schedule-related fields to DataExport API https://github.com/Rebilly/RebillyReportsAPI/pull/41

### CODE REVIEW

File: spec/definitions/DataExport.yaml [hunk 1]
```diff
@@ -14,6 +14,10 @@ properties:
     description: File ID
     readOnly: true
     type: string
+  userId:
+    description: User ID
```
Comment (adamaltman) [inline spec/definitions/DataExport.yaml]:
```
description: The ID of the User who requested the data export
```

File: spec/paths/data-exports@{id}.yaml [hunk 2]
```diff
@@ -28,3 +28,47 @@ get:
...
     401:
       $ref: "#/responses/AccessForbidden"
+    404:
+      $ref: "#/responses/NotFound"
+put:
+  tags:
+    - Data Exports
+  summary: Modify a data export
+  description: |
+    Modify an existing data export
```
Comment (adamaltman) [inline spec/paths/data-exports@{id}.yaml]:
```
description: |
  A data export may be changed as long as it is
  still pending.
```

File: spec/definitions/DataExport.yaml [hunk 3]
```diff
@@ -30,6 +30,11 @@ properties:
     description: The number of records in the export
     readOnly: true
     type: integer
+  scheduledTime:
+    type: string
+    readOnly: true
+    format: date-time
+    description: The date and time of the next scheduled Export
   createdTime:
     description: Export schedule created time
```
Comment (adamaltman) [inline spec/definitions/DataExport.yaml]: I would fix this language:

```
description: The time the data export was requested
```

File: spec/definitions/DataExport.yaml [hunk 4]
```diff
@@ -30,6 +30,11 @@ properties:
     description: The number of records in the export
     readOnly: true
     type: integer
+  scheduledTime:
+    type: string
+    readOnly: true
+    format: date-time
+    description: The date and time of the next scheduled Export
```
Comment (adamaltman) [inline spec/definitions/DataExport.yaml]: I would fix this language... this is not the correct.

```
description: The date and time this data export is scheduled to generate a file
```

File: spec/definitions/DataExport.yaml [hunk 5]
```diff
@@ -26,26 +30,21 @@ properties:
...
     description: The links related to resource
     readOnly: true
     minItems: 1
-    maxItems: 2
+    maxItems: 3
     items:
       - $ref: "#/definitions/SelfLink"
-      - $ref: "#/definitions/FileLink"
+      - $ref: "#/definitions/UserLink"
+      - $ref: "#/definitions/DownloadLink"
```
Comment (pmeth) [inline spec/definitions/DataExport.yaml]: we also need a signedLink

File: spec/definitions/DataExport.yaml [hunk 6]
```diff
@@ -26,26 +30,22 @@ properties:
...
+    description: Data export updated time
     allOf:
       - $ref: "#/definitions/ServerTimestamp"
   _links:
     type: array
     description: The links related to resource
     readOnly: true
     minItems: 1
-    maxItems: 2
+    maxItems: 3
```
Comment (pmeth) [inline spec/definitions/DataExport.yaml]: should be 4 I think

File: spec/definitions/DataExport.yaml [hunk 7]
```diff
@@ -14,6 +14,10 @@ properties:
     description: File ID
```
Comment (adamaltman) [inline spec/definitions/DataExport.yaml]: I believe file id should be removed.

**Comment by adamaltman** [comment]
We should eliminate `dataExportInstruction` and promote all members inside to the top level.

**Pull request merged**